whole-tale/dashboard
# Migrate "View/Edit Tale Metadata" functionality to the Run view

Changes:
* Added Interact / Metadata tabs to the "Run" view
    * Interact Tab: the previous "Run" view can be found here - iframe or link to the wholetale frontend/instance
    * Metadata Tab: a new view based heavily on the previous "View/Edit Tale" view can be found here


Test Case

1. Checkout this branch locally, pointed at `girder.dev.wholetale.org`: `git fetch --all && git checkout refactor-view-edit-tale-metadata`
2. Start Ember: `ember s --environment=production`
3. Login to the dev instance that you just started
4. Launch a Tale that you own - create one if needed
5. Select the Tale Instance that you just launched and choose "Run" at the top
    * You should see the "Interact" and "Metadata" tabs
    * The Interact tab should be selected by default, which should show the familiar Tale view (e.g. iframe or link)
6. Click the "Metadata" tab
    * Tab content should change to offer "View/Edit Tale Metadata" functionality
    * If you do not own or cannot edit the Tale, the form should be read-only
    * Even if you do own the Tale, `Date Created`, `Last Updated`, and `Authors` should still be read-only
    * Valid environments should be listed in the form dropdown for `Environment`
7. Change the name of the Tale and click "Save"
    * NOTE: The Environment field will not currently save - see https://github.com/whole-tale/dashboard/issues/285#issuecomment-440426589
8. Refresh the page
    * Your changes should be preserved between page refreshes, confirming that they are being properly persisted to the server
